src/sync.h

```diff
@@ -105,35 +115,58 @@ namespace librealsense
             // wait until user is done with all the stuff he chose to borrow
             _callback_inflight.wait_until_empty();
         }
+       std::string get_name()
```

Comment:
Consider refactoring out the `pure virtual` parts and the shared code to separate classes.
`mtacher_interface` to hold the interface (pure virtual only).
and the shared code to either a `matcher_base` or IMO better yet to a class or classes that provides those functionalities.
It will make the code cleaner, simpler to read and have a stronger static enforcement (e.g - `_name` and `_active` might not be initialized by a deriving classes in the current form)